Remove minor build warnings

hugueskamba opened the pull request:
### Description
The various warnings removed were observed when building `mbed-os-example-blinky` for the `K64F` target with the `GCC_ARM` toolchain.

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [X] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers
@kjbracey-arm @evedon 
<!--
    Optional
    Request additional reviewers with @username
-->

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

hugueskamba committed:
Remove minor build warnings

SeppoTakalo commented on features/frameworks/mbed-coap/source/sn_coap_protocol.c:
```diff
@@ -1203,6 +1203,8 @@ void sn_coap_protocol_linked_list_duplication_info_remove(struct coap_s *handle,
 #endif //SN_COAP_DUPLICATION_MAX_MSGS_COUNT
 }
+
+#if SN_COAP_DUPLICATION_MAX_MSGS_COUNT
```
Mbed Coap comes from external source.
@mikter or @artokin Is this acceptable to be modified here?

0xc0170 replied:
> @mikter or @artokin Is this acceptable to be modified here?

Any update? Can this go in or not?

artokin replied:
All other references to sn_coap_protocol_duplication_info_free are already flagged by SN_COAP_DUPLICATION_MAX_MSGS_COUNT and therefore this is good to go.

CC: @anttiylitokola

hugueskamba committed:
Use PRIu32 to print uint32_t variable

As the variable underlying type size is different depending on the
toolchain used

0xc0170 commented on components/TARGET_PSA/services/attestation/COMPONENT_PSA_SRV_IMPL/tfm_impl/attestation_core.c:
```diff
@@ -869,7 +869,7 @@ attest_create_token(struct useful_buf_c *challenge,
                                    option_flags,         /* option_flags */
                                    key_select,           /* key_select   */
-                                   COSE_ALGORITHM_ES256, /* alg_select   */
+                                   alg_select,           /* alg_select   */
```
@ARMmbed/mbed-os-crypto This one also should be fixed in tfm repository?

Patater replied:
Yes, please https://git.trustedfirmware.org/trusted-firmware-m.git/tree/secure_fw/services/initial_attestation/attestation_core.c#n918

hugueskamba replied:
The code in the trusted firmware repo is different than the version in Mbed OS.
The variable `alg_select` has been added to the Mbed OS version along with 
```
    /* Map the key select to an algorithm. Maybe someday we'll support something
    * other than ES256
   */
    switch (key_select) {
    default:
        alg_select = COSE_ALGORITHM_ES256;
    }
```
Therefore no fix is necessary upstream.

Is the file usually synchronized with the upstream version? If so, should the Mbed OS version of the file (well the function at least) be changed instead to mirror its upstream counterpart?

Patater replied:
OK, agreed that no change upstream is necessary.

Yes, we should take the upstream version when we replace our attestation service with the TF-M one.

In the mean time, I'd suggest removing the switch to fix the warning, as that keeps the code better aligned with upstream TF-M.

hugueskamba replied:
9f44c2c

hugueskamba committed:
Remove extraneous parentheses.

0xc0170 reviewed (changes requested):
Waiting for crypto and mesh team to review

hugueskamba committed:
Sync with TF upstream code as much as possible to remove warning

0xc0170 reviewed (approved):
All changes approved
